Formula/cmake.rb

```diff
@@ -46,6 +52,9 @@ def install
       --system-bzip2
       --system-curl
     ]
+    on_linux do
+      args -= ["--system-zlib", "--system-bzip2", "--system-curl"]
```

Comment:
It's probably not a really big deal, but stylistically, would it make more sense to remove these args from the preceding array and then add them in an `on_macos` block? To me at least, it feels less weird to conditionally add the args than to add them and then conditionally remove them.